app/dao/status.go

```diff
@@ -0,0 +1,48 @@
+package dao
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"yatter-backend-go/app/domain/object"
+	"yatter-backend-go/app/domain/repository"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type (
+	// Implementation for repository.Status
+	status struct {
+		db *sqlx.DB
+	}
+)
+
+// Create status repository
+func NewStatus(db *sqlx.DB) repository.Status {
+	return &status{db: db}
+}
+
+// FindById : StatusIdからstatusを取得
+func (r *status) FindById(ctx context.Context, id object.StatusID) (*object.Status, error) {
```

Comment:
```suggestion
func (r *status) FindByID(ctx context.Context, id object.StatusID) (*object.Status, error) {
```
`id` は go の命名規則では `ID` となります。他にはURLなどのように短縮系が一般的になっているものに関しては全て大文字になります！
https://github.com/golang/go/wiki/CodeReviewComments#initialisms